From 327c3dcbfa005f5d1b2fcda07ca1198d65bd23c5 Mon Sep 17 00:00:00 2001
From: Hongxu Jia <hongxu.jia@windriver.com>
Date: Wed, 11 Oct 2017 15:40:42 +0800
Subject: [PATCH 1/7] Makefile.vulkan.am: explictly add lib expat to intel
 libvulkan's lib depends
Organization: O.S. Systems Software LTDA.

While built with "-fvisibility=default"
...
|i586-oe-linux-gcc ... -fvisibility=default ... -o common/.libs/common_libintel_common_la-gen_decoder.o
...

It triggered the failure
...
|i586-oe-linux-g++  ... common/.libs/libintel_common.a ... -o vulkan/.libs/libvulkan_intel.so
|common/.libs/libintel_common.a(common_libintel_common_la-gen_decoder.o):
|In function `start_element':
|/usr/src/debug/mesa/2_17.1.7-r0/mesa-17.1.7/src/intel/common/gen_decoder.c:371:
undefined reference to `XML_GetCurrentLineNumber'
...

explictly add EXPAT_LIBS to intel's VULKAN_LIB_DEPS

Upstream-Status: Accepted
https://cgit.freedesktop.org/mesa/mesa/commit/?id=05fc62d89f59ce19a18bfd4e63a09624910d6caf

Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>
---
 src/intel/Makefile.vulkan.am | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/intel/Makefile.vulkan.am b/src/intel/Makefile.vulkan.am
index 4125cb205a..356c4f8452 100644
--- a/src/intel/Makefile.vulkan.am
+++ b/src/intel/Makefile.vulkan.am
@@ -161,6 +161,7 @@ VULKAN_LIB_DEPS = \
 	$(LIBDRM_LIBS) \
 	$(PTHREAD_LIBS) \
 	$(DLOPEN_LIBS) \
+	$(EXPAT_LIBS) \
 	-lm
 
 if HAVE_PLATFORM_ANDROID
-- 
2.17.1

